apps/drec-api/src/pods/integrators/integrators.service.ts

```diff
@@ -0,0 +1,194 @@
+import { Injectable, Logger, Inject } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { DateTime } from 'luxon';
+import {
+  MeasurementDTO,
+  ReadDTO,
+  Unit,
+  ReadsService as BaseReadsService,
+} from '@energyweb/energy-api-influxdb';
+import { BASE_READ_SERVICE } from '../reads/const';
+import { DeviceService } from '../device/device.service';
+import { HttpService } from '@nestjs/axios';
+import { Integrator } from '../../utils/enums';
+import { ConfigService } from '@nestjs/config';
+import FormData from 'form-data';
+import { map, catchError } from 'rxjs/operators';
+import { throwError } from 'rxjs';
+import { AxiosRequestConfig } from 'axios';
+import { DeviceDTO } from '../device/dto';
+import { GenerationReadingStoredEvent } from '../../events/GenerationReadingStored.event';
+import { EventBus } from '@nestjs/cqrs';
+import { BigNumber } from 'ethers';
+
+@Injectable()
+export class IntegratorsService {
+  private readonly logger = new Logger(IntegratorsService.name);
+
+  constructor(
+    private httpService: HttpService,
+    private deviceService: DeviceService,
+    @Inject(BASE_READ_SERVICE)
+    private baseReadsService: BaseReadsService,
+    private readonly configService: ConfigService,
+    private readonly eventBus: EventBus,
+  ) {}
+
+  // @Cron(CronExpression.EVERY_30_SECONDS)
+  @Cron('0 30 01 * * *') // Every day at 01:30 - Server Time
+  async handleBBOXcron(): Promise<void> {
+    this.logger.debug('BBOX Cron called every day at 01:30 - Server Time');
+
+    const integrator: Integrator = Integrator.BBOX;
+    const startDate = DateTime.now()
+      .minus({ days: 1 })
+      .toUTC()
+      .toFormat('yyyy-MM-dd HH:mm:ss');
+    const endDate = DateTime.now()
+      .minus({ minute: 1 })
+      .toUTC()
+      .toFormat('yyyy-MM-dd HH:mm:ss');
+
+    this.logger.debug(`Start date ${startDate} - End date ${endDate}`);
+
+    const devices = await this.deviceService.findMultiple({
+      where: {
+        integrator,
+      },
+    });
```

Comment:
It is better to decouple our services from typeorm. You can consider refactoring this to use our interface instead of FindOptions.

Reply:
Not sure exactly if I understand what you mean